### ISSUE TITLE
Prototype based inheritance for childs

### ISSUE THREAD

**Issue by StarpTech** [delvedor/workq#4]
What do you think when we could set the `dry` handler selectively so that the event is bubbling when no handler was defined?
What do you think when we could add attributes to the `this` instance without to modify other child's but preserving the standard API?

Those concepts are extensively used in fastify and hemerajs and easy to solve with `Object.create`
I think this wouldn't bloat the API but providing a great level of isolation.

**Comment by StarpTech** [comment]
You can also limit the api on childs (second argument of Object.create) e.g to hide private variables.

**Comment by delvedor** [comment]
> What do you think when we could set the dry handler selectively so that the event is bubbling when no handler was defined?

Can you rephrase?

> What do you think when we could add attributes to the this instance without to modify other child's but preserving the standard API?

I think this is out of scope of the library :P

**Comment by StarpTech** [comment]
I just realized the whole concept. We create a new queue for each child. If you have multiple jobs on the root level the `dry` event is called multiple times so each level can fire the `dry` event independently. That's fine to me and I can close the issue :)
